Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -258,12 +273,35 @@ protected override void ProcessRecord()
             string objContentType = null;
             Stream contentStream = null;
 
+            // Check whether our current location is in gs:\ (i.e., we are in the Google Cloud Storage provider).
+            // If so, then we can resolve the path to get the bucket and folder name (if we are in one).
+            if (SessionState?.Path?.CurrentLocation?.Provider?.ImplementingType == typeof(GoogleCloudStorageProvider))
```

Comment:
Might be worth checking if bucket and ObjectNamePrefix are set already to avoid doing unneeded work